sphinx/screens/chats/chat-tribe/chat-tribe/src/main/java/chat/sphinx/chat_tribe/ui/ChatTribeViewModel.kt

```diff
@@ -271,6 +310,26 @@ internal class ChatTribeViewModel @Inject constructor(
                 _feedDataStateFlow.value = TribeFeedData.Result.NoFeed
             }
         }
+
+        pinJob?.cancel()
+        pinJob = viewModelScope.launch(mainImmediate) {
+            chatRepository.getChatById(chatId).firstOrNull()?.let { chat ->
+                chat.pinedMessage?.let { messageId ->
+                    messageRepository.getMessageById(messageId).firstOrNull()?.let { message ->
+                        pinedMessageHeaderViewState.updateViewState(
+                            PinedMessageHeaderViewState.PinedMessageHeader(
+                                message
+                            )
+                        )
+
+                    }
+                } ?: run {
+                    pinedMessageHeaderViewState.updateViewState(
+                        PinedMessageHeaderViewState.Idle
+                    )
+                }
+            }
+        }
```

Comment:
is this where we're parsing the data to see which message to pin?

Just trying to understand the code a bit

Reply:
yes it is, it should work fine on the tribe member side
i implemented this to retrieve the pin message both for owner& member 
when ever the chat opens and the pin field is not empty it shows the pin message

Reply:
This is where it searches for the Message db record of the pinned message and it shows the bar at the top if chat has a pinned message and that message is in the db.